src/auth/auth.ts

```diff
@@ -0,0 +1,102 @@
+import { TOTPNamespace } from "./cognito/types/totpNameSpace";
+import { AuthDevice } from "./types/authDevice";
+import { AuthSession } from "./types/authSession";
+import { AuthCodeDeliveryDetails, AuthUserAttribute } from "./types/models";
+import {
+  ConfirmResetPasswordRequest,
+  ConfirmSignInRequest,
+  ConfirmSignUpRequest,
+  ConfirmUserAttributeRequest,
+  FetchAuthSessionRequest,
+  FetchUserAttributesRequest,
+  ForgetDeviceRequest,
+  ResendSignUpCodeRequest,
+  ResendUserAttributeConfirmationCodeRequest,
+  ResetPasswordRequest,
+  SetPreferredMFARequest,
+  SignInRequest,
+  SignInWithWebUIRequest,
+  SignOutRequest,
+  SignUpRequest,
+  UpdatePasswordRequest,
+  UpdateUserAttributeRequest,
+  UpdateUserAttributesRequest,
+} from "./types/request";
+import { GetCurrentUserRequest } from "./types/request/getCurrentUserRequest";
+import {
+  AuthSignInResult,
+  AuthSignUpResult,
+  AuthUser,
+  ResetPasswordResult,
+  SignOutResult,
+  UpdateUserAttributesResult,
+} from "./types/result";
+import { UpdateUserAttributeResult } from "./types/result/updateUserAttributeResult";
+import { AuthSignInStep } from "./types/step";
+
+export interface AuthInterface {
```

Comment:
Mentioned for the API PR as well, but I wonder if we should declare functions directly or use a namespace to indicate that we're moving away from these monolithic classes? It looks like we already have this in the cognito commands type defs below, that might be sufficient for the purposes of the RFC.

Reply:
Thanks Jim for reviewing this PR. Yes I think we'd like to get feedback on API usage, specially on response and error handling. Some APIs offer a `nextStep` that provides information on how to proceed with the authentication process.